source/create.ts

```diff
@@ -56,7 +56,7 @@ interface GotFunctions {
 
 export interface Got extends Record<HTTPAlias, GotFunctions>, GotFunctions {
 	stream: GotStream;
-	defaults: Defaults | Readonly<Defaults>;
+	defaults: Defaults;
```

Comment:
No point in having `Defaults | Readonly<Defaults>` as in some cases TS misguesses it as `Readonly<Defaults>` and people have to cast it to `Defaults` anyway. The JS runtime will throw anyway if anyone tries to modify frozen defaults.